src/main/kotlin/cacophony/semantic/syntaxtree/SyntaxTree.kt

```diff
@@ -112,14 +112,34 @@ sealed class Definition(
                 areEquivalentTypes(type, other.type)
     }
 
-    class FunctionDeclaration(
+    sealed class FunctionDef(
         range: Pair<Location, Location>,
         identifier: String,
         val type: Type.Functional?,
-        val arguments: List<FunctionArgument>,
         val returnType: Type,
+    ) : Definition(range, identifier) {
+        override fun isEquivalent(other: Expression?): Boolean =
+            super.isEquivalent(other) &&
+                other is FunctionDeclaration &&
+                areEquivalentTypes(type, other.type) &&
+                areEquivalentTypes(returnType, other.returnType)
+    }
+
+    class ForeignFunctionDef(
+        range: Pair<Location, Location>,
+        identifier: String,
+        type: Type.Functional,
+        returnType: Type,
+    ) : FunctionDef(range, identifier, type, returnType), TreeLeaf
```

Comment:
We can't have own function declarations in code, but we can have foreignFunction declarations, so maybe we can add toString for printing. I'm unsure if foreign function declarations will be present in AST, so I don't know if my comment makes sense.

Reply:
Yes, definitely, I added both this and isEquivalent